internal/storage/v1/elasticsearch/spanstore/writer.go

```diff
@@ -103,12 +105,37 @@ type spanAndServiceIndexFn func(spanTime time.Time) (string, string)
 
 func getSpanAndServiceIndexFn(p SpanWriterParams, writeAlias string) spanAndServiceIndexFn {
 	spanIndexPrefix := p.IndexPrefix.Apply(spanIndexBaseName)
+	hasSpanOverride := p.SpanIndexOverride != ""
+	if hasSpanOverride {
+		spanIndexPrefix = p.SpanIndexOverride
+	}
```

Comment:
why is this needed? There is a very simple function in L133 `if p.UseReadWriteAliases {` which covers the overrides condition and where overrides can be applied instead of composing index name from pieces